stix2/pattern_expressions.py

```diff
@@ -0,0 +1,223 @@
+
+    def __str__(self):
+        # if isinstance(self.rhs, list):
+        #     final_rhs = []
+        #     for r in self.rhs:
+        #         final_rhs.append("'" + self.escape_quotes_and_backslashes("%s" % r) + "'")
+        #     rhs_string = "(" + ", ".join(final_rhs) + ")"
+        # else:
+        #     rhs_string = self.rhs
+        if self.negated:
+            return "%s NOT %s %s" % (self.lhs, self.operator, self.rhs)
+        else:
+            return "%s %s %s" % (self.lhs, self.operator, self.rhs)
+
+
+class EqualityComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(EqualityComparisonExpression, self).__init__("=", lhs, rhs, negated)
+
+
+class GreaterThanComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(GreaterThanComparisonExpression, self).__init__(">", lhs, rhs, negated)
+
+
+class LessThanComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(LessThanComparisonExpression, self).__init__("<", lhs, rhs, negated)
+
+
+class GreaterThanEqualComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(GreaterThanComparisonExpression, self).__init__(">=", lhs, rhs, negated)
+
+
+class LessThanEqualComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(LessThanComparisonExpression, self).__init__("<=", lhs, rhs, negated)
+
+
+class InComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(InComparisonExpression, self).__init__("IN", lhs, rhs, negated)
+
+
+class LikeComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(LikeComparisonExpression, self).__init__("LIKE", lhs, rhs, negated)
+
+
+class MatchesComparisonExpression(ComparisonExpression):
+    def __init__(self, lhs, rhs, negated=False):
+        super(MatchesComparisonExpression, self).__init__("MATCHES", lhs, rhs, negated)
+
+
+class IsSubsetComparisonExpression(ComparisonExpression):
+        def __init__(self, lhs, rhs, negated=False):
+            super(IsSubsetComparisonExpression, self).__init__("ISSUBSET", lhs, rhs, negated)
+
+
+class IsSupersetComparisonExpression(ComparisonExpression):
+        def __init__(self, lhs, rhs, negated=False):
+            super(IsSupersetComparisonExpression, self).__init__("ISSUPERSET", lhs, rhs, negated)
+
+
+class BooleanExpression(PatternExpression):
+    def __init__(self, operator, operands):
+        self.operator = operator
+        self.operands = []
+        for arg in operands:
+            if not hasattr(self, "root_type"):
+                self.root_type = arg.root_type
+            elif self.root_type and (self.root_type != arg.root_type) and operator == "AND":
+                raise ValueError("This expression cannot have a mixed root type")
```

Comment:
I think this error message would be clearer if it said something like `All operands to an "AND" expression must have the same object type`.